src/libpgmoneta/wal.c

```diff
@@ -285,6 +285,7 @@ pgmoneta_wal(int srv, char** argv)
                            pgmoneta_log_error("Could not create or open WAL segment file at %s", d);
                            goto error;
                         }
+                        snprintf(config->servers[srv].current_wal_filename, MISC_LENGTH, "%s.partial", filename);
```

Comment:
Memset the buffer to 0 before you populate it. The filename is of the same length so this should be fine, but just in case:)